protocol-specs/generators/transaction_network_id_and_change_order/index.js

```diff
@@ -0,0 +1,609 @@
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const {
+	hexToBuffer,
+	intToBuffer,
+	hash,
+	getFirstEightBytesReversed,
+	bufferToIntAsString,
+	signData,
+} = require('@liskhq/lisk-cryptography');
+const BaseGenerator = require('../base_generator');
+
+const accounts = [
+	{
+		passphrase:
+			'wear protect skill sentence lift enter wild sting lottery power floor neglect',
+		privateKey:
+			'8f41ff1e75c4f0f8a71bae4952266928d0e91660fc513566ac694fed61157497efaf1d977897cb60d7db9d30e8fd668dee070ac0db1fb8d184c06152a8b75f8d',
+		publicKey:
+			'efaf1d977897cb60d7db9d30e8fd668dee070ac0db1fb8d184c06152a8b75f8d',
+		address: '2129300327344985743L',
+	},
+	{
+		passphrase:
+			'inherit moon normal relief spring bargain hobby join baby flash fog blood',
+		privateKey:
+			'de4a28610239ceac2ec3f592e36a2ead8ed4ac93cb16aa0d996ab6bb0249da2c0b211fce4b615083701cb8a8c99407e464b2f9aa4f367095322de1b77e5fcfbe',
+		publicKey:
+			'0b211fce4b615083701cb8a8c99407e464b2f9aa4f367095322de1b77e5fcfbe',
+		address: '18141291412139607230L',
+	},
+	{
+		passphrase:
+			'better across runway mansion jar route valid crack panic favorite smooth sword',
+		privateKey:
+			'de1520f8589408e76a97643ba7d27f20009b06899816c8af20f9b03f4a4bd8a66766ce280eb99e45d2cc7d9c8c852720940dab5d69f480e80477a97b4255d5d8',
+		publicKey:
+			'6766ce280eb99e45d2cc7d9c8c852720940dab5d69f480e80477a97b4255d5d8',
+		address: '13191770412077040757L',
+	},
+	{
+		passphrase:
+			'mirror swap middle hunt angle furnace maid scheme amazing box bachelor debris',
+		privateKey:
+			'ad7462eb8f682b0c3424213ead044381ba0007bb65ce26287fc308027c871d951387d8ec6306807ffd6fe27ea3443985765c1157928bb09904307956f46a9972',
+		publicKey:
+			'1387d8ec6306807ffd6fe27ea3443985765c1157928bb09904307956f46a9972',
+		address: '2443122499609067441L',
+	},
+];
+
+const networkIdentifier =
+	'e48feb88db5b5cf5ad71d93cdcd1d879b6d5ed187a36b0002cc34e0ef9883255';
+
+const getId = transactionBytes => {
+	const transactionHash = hash(transactionBytes);
+	const bufferFromFirstEntriesReversed = getFirstEightBytesReversed(
+		transactionHash,
+	);
+	const transactionId = bufferToIntAsString(bufferFromFirstEntriesReversed);
+
+	return transactionId;
+};
+
+const generateValidTransferTransaction = () => {
+	const tx = {
+		type: 8,
+		senderPublicKey: accounts[0].publicKey,
+		timestamp: 54316324,
+		asset: {
+			recipientId: accounts[1].address,
+			amount: '1234567890',
+			data: 'random data',
+		},
+	};
+
+	const transactionTimestamp = Buffer.alloc(4);
+	transactionTimestamp.writeIntLE(tx.timestamp, 0, 4);
+	const txBuffer = Buffer.concat([
+		Buffer.alloc(1, tx.type),
+		transactionTimestamp,
+		hexToBuffer(tx.senderPublicKey),
+		intToBuffer(tx.asset.amount, 8, 'big'),
+		intToBuffer(tx.asset.recipientId.slice(0, -1), 8),
+		Buffer.from(tx.asset.data, 'utf8'),
+	]);
+
+	const signature = signData(
+		hash(Buffer.concat([hexToBuffer(networkIdentifier), txBuffer])),
```

Comment:
This is not a final implementation, but just to generate the source of truth.
`signData` is generic cryptographic function, and it will not be changed because it is used not only for signing the transaction